packages/backend/src/modules/finality/analyzers/arbitrum/getSegments.ts

```diff
@@ -0,0 +1,39 @@
+import { assert } from '@l2beat/shared-pure'
+import zlib from 'zlib'
+
+import { rlpDecode } from '../../utils/rlpDecode'
+import { byteArrFromHexStr } from '../opStack/utils'
+import { blobsToData } from './blobsToData'
+import { numberToByteArr } from './utils'
+
+export function getSegments(relevantBlobs: { blob: string }[]) {
+  const blobs = relevantBlobs.map(({ blob }) => byteArrFromHexStr(blob))
+  const payload = blobsToData(blobs)
+  const decompressed = decompressPayload(payload)
+  const rlpEncoded = concatWithLengthForRlp(decompressed)
+  const segments = rlpDecode(rlpEncoded)
+  assert(Array.isArray(segments), 'Expected segments to be an array')
+
+  return segments
+}
+
+export function decompressPayload(payload: Uint8Array): Uint8Array {
+  // todo: we should check the first byte. It's probably the compression type.
```

Comment:
It is. Should always be `0x00` for brotli compression

Reply:
That's why I like your reviews

Reply:
[b68a382](https://github.com/l2beat/l2beat/pull/3288/commits/b68a3822f6b699bbcb8be92ea966c620f0e0d88a)